sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuMultiFileBatchReader.java

```diff
@@ -0,0 +1,252 @@
+  private static final Logger LOG = LoggerFactory.getLogger(GpuMultiFileBatchReader.class);
+  private final Map<String, Tuple2<Map<Integer, ?>, Schema>> constsSchemaMap =
+      Maps.newConcurrentMap();
+  private final LinkedHashMap<String, FileScanTask> files;
+  private final Schema expectedSchema;
+  private final boolean caseSensitive;
+  private final Configuration conf;
+  private final int maxBatchSizeRows;
+  private final long maxBatchSizeBytes;
+  private final String parquetDebugDumpPrefix;
+  private final scala.collection.immutable.Map<String, GpuMetric> metrics;
+  private final boolean useMultiThread;
+  private final FileFormat fileFormat;
+  private final int numThreads;
+  private final int maxNumFileProcessed;
+
+  private NameMapping nameMapping = null;
+  private boolean needNext = true;
+  private boolean isBatchPending;
+  // lazy variables
+  private FilePartitionReaderBase rapidsReader = null;
+
+  GpuMultiFileBatchReader(CombinedScanTask task, Table table, Schema expectedSchema,
+      boolean caseSensitive, Configuration conf, int maxBatchSizeRows, long maxBatchSizeBytes,
+      String parquetDebugDumpPrefix, int numThreads, int maxNumFileProcessed,
+      boolean useMultiThread, FileFormat fileFormat,
+      scala.collection.immutable.Map<String, GpuMetric> metrics) {
+    super(table, task);
+    this.expectedSchema = expectedSchema;
+    this.caseSensitive = caseSensitive;
+    this.conf = conf;
+    this.maxBatchSizeRows = maxBatchSizeRows;
+    this.maxBatchSizeBytes = maxBatchSizeBytes;
+    this.parquetDebugDumpPrefix = parquetDebugDumpPrefix;
+    this.useMultiThread = useMultiThread;
+    this.fileFormat = fileFormat;
+    this.metrics = metrics;
+    this.numThreads = numThreads;
+    this.maxNumFileProcessed = maxNumFileProcessed;
+    String nameMapping = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+    if (nameMapping != null) {
+      this.nameMapping = NameMappingParser.fromJson(nameMapping);
+    }
+    files = Maps.newLinkedHashMapWithExpectedSize(task.files().size());
+    task.files().forEach(fst -> this.files.putIfAbsent(fst.file().path().toString(), fst));
+  }
+
+  @Override
+  public ColumnarBatch get() {
+    if (rapidsReader == null) {
+      // Not initialized, return null to align with PerFile reader.
+      return null;
+    }
+    needNext = true;
+    isBatchPending = false;
+    // The same post-process with PerFile reader.
+    try (ColumnarBatch batch = rapidsReader.get()) {
+      // The Rapids reader should already set the current file.
+      String curFile = InputFileUtils.getCurInputFilePath();
+      // <idToConstants, updatedExpectedSchema>
+      Tuple2<Map<Integer, ?>, Schema> constsSchema = constsSchemaMap.get(curFile);
+      return GpuIcebergReader.addUpcastsIfNeeded(
+          GpuIcebergReader.addConstantColumns(batch, constsSchema._2(), constsSchema._1()),
+          constsSchema._2());
```

Comment:
Some local variables can help with readability here.
```suggestion
      Tuple2<Map<Integer, ?>, Schema> constsSchema = constsSchemaMap.get(curFile);
      Map<Integer, ?> idToConstants = constsSchema._1();
      Schema updatedExpectedSchema = constsSchema._2();
      return GpuIcebergReader.addUpcastsIfNeeded(
          GpuIcebergReader.addConstantColumns(batch, updatedExpectedSchema, idToConstants),
          updatedExpectedSchema);
```

Reply:
updated.